src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -4235,6 +4255,112 @@ public void kEnumTest() throws DatabaseEngineException {
                 .isEqualTo(TestEnum.TEST_ENUM_VAL.name());
     }
 
+    /**
+     * Tests that when inserting duplicated entries in a table the right exception is returned.
+     *
+     * The steps performed on this test are:
+     * <ol>
+     *     <li>Add duplicated entries in a transaction and fail to persist</li>
+     *     <li>Ensure the existence of the Exception and rollback transaction</li>
+     *     <li>Ensure the exception is a {@link DatabaseEngineUniqueConstraintViolationException}</li>
+     * </ol>
+     *
+     * @throws DatabaseEngineException If there is a problem on {@link DatabaseEngine} operations.
+     */
+    @Test
+    public void insertDuplicateDBError() throws Exception {
+        create5ColumnsEntityWithPrimaryKey();
+
+        engine.beginTransaction();
+
+        DatabaseEngineException expectedException = null;
+        try {
+            EntityEntry entry = entry().set("COL1", 2).set("COL2", false).set("COL3", 2D).set("COL4", 3L).set("COL5", "ADEUS")
+                                       .build();
+            // Add the same entry twice (repeated value for COL1, id)
+            engine.persist("TEST", entry);
+            engine.persist("TEST", entry);
+
+            fail("Was expecting the persist operation to fail");
+        } catch (final DatabaseEngineException e) {
+            expectedException = e;
+        } finally {
+            if (engine.isTransactionActive()) {
+                engine.rollback();
+            }
+        }
+
+        // Ensure we had an exception and therefore we didn't insert anything on the DB and that we cleared the batches.
+        assertNotNull("DB returned exception when flushing", expectedException);
+
+        // Ensure that the exception thrown is caused by an SQL error.
+        assertThat(expectedException.getCause())
+                .as("Encapsulated exception is SQLException")
+                .isInstanceOf(SQLException.class);
+
+        // Ensure that the exception matches a unique constraint violation and thus is a DatabaseEngineUniqueConstraintViolationException.
+        assertThat(expectedException)
+                .as("Is unique constraint violation exception")
+                .isInstanceOf(DatabaseEngineUniqueConstraintViolationException.class);
+
+        // Ensure that the exception is caught in AbstractDatabaseEngine#persist and then handled in QueryExceptionHandler#handleException.
+        assertEquals("Something went wrong persisting the entity [unique_constraint_violation]", expectedException.getMessage());
```

Comment:
No need to try-catch the exception, and some other things...

1. the first _persist_ should not fail; if it does, its exception should not be caught, and the test should fail immediately
2. no need to catch the exception and analyze it later all the assertions can be done with the proper method of AssertJ :smile_cat: 

```suggestion
        try {
            // Add the same entry twice (repeated value for COL1, id)
            engine.persist("TEST", entry);

            assertThatCode(() -> engine.persist("TEST", entry))
                    .as("Is unique constraint violation exception")
                    .isInstanceOf(DatabaseEngineUniqueConstraintViolationException.class)
                    .as("Encapsulated exception is SQLException")
                    .hasCauseInstanceOf(SQLException.class)
                    .hasMessage("Something went wrong persisting the entity [unique_constraint_violation]");

        } finally {
            if (engine.isTransactionActive()) {
                engine.rollback();
            }
        }
```

Reply:
(same applies to the next `batchInsertDuplicateDBError` test)